hpy/debug/src/debug_ctx_cpython.c

```diff
@@ -78,8 +78,24 @@ void debug_ctx_CallRealFunctionFromTrampoline(HPyContext *dctx,
         HPyFunc_noargs f = (HPyFunc_noargs)func;
         _HPyFunc_args_NOARGS *a = (_HPyFunc_args_NOARGS*)args;
         DHPy dh_self = _py2dh(dctx, a->self);
-        DHPy dh_result = f(dctx, dh_self);
-        DHPy_close_and_check(dctx, dh_self);
+
+        HPyContext *next_dctx = hpy_debug_get_next_dctx_from_cache(dctx);
```

Comment:
That's a lot of code duplication. I think we could pull the _temporary debug context setup and tear down code_ out of the switch if we use `break` instead of `return` at the end of the cases.
We then only need to use `next_dctx` for opening and closing the argument handles like `dh_self`.
Or do I miss something?

Reply:
Fixed. We cannot really pull out the whole thing, because in `a->result` variable `a` is a different struct every time, but I pulled as much as I could. Also used the new helpers in the autogenerated code.

Reply:
lgtm